### ISSUE TITLE
Return focus on outside click does not respect "returnFocusOnDeactivate" value

### ISSUE THREAD

**Issue by Tckool** [focus-trap/focus-trap#103]
After this issue: https://github.com/davidtheclark/focus-trap/issues/60, the click handler returns the focus to the focus trap if the new target is not clickable. However, this happens even if the `returnFocusOnDeactivate` configuration option is passed in as false: https://github.com/davidtheclark/focus-trap/blob/master/index.js#L111-L114

Because I have set `returnFocusOnDeactivate` to false, I was expecting that the focus trap would not return the focus. I feel like it should respect that configuration value and only return the focus when it is set to true and the new target is not clickable. 

It was confusing for me so I am just wondering if this is intentional or if it is a bug.  If it is intentional, I'm curious as to why, and if it is just a bug, I would be happy to open a PR to fix it.

**Comment by Slapbox** [comment]
@davidtheclark @stefcameron any thoughts on this? `returnFocusOnDeactivate` seems to have no effect no matter where I try to move my focus to from the focus-trap.

I've tried several configurations and it seems impossible to workaround. This library is really fantastic but if I can't find a way to workaround this then I'll have to abandon focus-trap which would be a damn shame because it's so great otherwise.

```js
                  <FocusTrap
                    returnFocusOnDeactivate={false}
                    allowOutsideClick // Added trying to get returnFocusOnDeactivate={false} to work as expected
                    clickOutsideDeactivates  // Added trying to get returnFocusOnDeactivate={false} to work as expected
                  >
                      {children}
                  </FocusTrap>
```

**Comment by stefcameron** [comment]
@Slapbox Thanks for bringing this issue to my attention! I recently took over as maintainer for this repo, so I'm not familiar with everything just yet.

I took some time to look at how #60 was fixed, and while the commit had lots of changes in it, the essence of the commit comes down to this: https://github.com/focus-trap/focus-trap/blob/master/index.js#L230

That change is causing what I would consider a __bug__. I believe the intention of the change, to fix #60, was to state that if the outside element that was clicked is focusable, then _do not_ "return focus" (to the element that was focused when the trap was activated), letting that new outside element get the focus as a result of _not_ preventing default on the click event that the trap captured. However, if the outside element clicked is __not__ focusable, then _do_ "return focus" (to the element that was focused when the trap was activated) -- but that's where we're missing the respect for the trap's `returnFocusOnDeactivate` option.

We respect that option in https://github.com/focus-trap/focus-trap/blob/master/index.js#L108-L111, but only if `returnFocus` is not defined (or no `deactivateOptions` are given). Unfortunately, because of the way the fix was made, `deactivateOptions.returnFocus` is always defined and a boolean.

So I would view this as a bug that we should definitely be fixing!

If I were to fix it, I would do so by updating https://github.com/focus-trap/focus-trap/blob/master/index.js#L230 to be `returnFocus: config.returnFocusOnDeactivate && !isFocusable(e.target)`

If you agree with my assessment, I'm open to a PR to fix it!

Bonus points for adding a [demo](https://github.com/focus-trap/focus-trap/tree/master/demo) and [test](https://github.com/focus-trap/focus-trap/blob/master/cypress/integration/focus-trap-demo.spec.js) to check it! I'm working to add test coverage to this library to hopefully prevent these kinds of bugs in the future.

**Comment by Slapbox** [comment]
This is totally a my bad. Very sorry to waste your time.

 I realized after like twenty minutes that I wasn't looking at the docs for the React version and that I'd somehow closed that tab and ended up on the normal focus-trap repo. So I was passing the options directly to the component rather than passing them under `focusTrapOptions`. ::facepalm::

I had thought I cleaned up my errant comments. Sorry again.

----

I do think the `returnFocusOnDeactivate` didn't work as I'd expect when I did realize what my mistake was. But because we're already handling focus very intentionally, I moved to using our internal solutions via the `onActivate` `onDeactivate` callbacks, so I can't say for sure.

**Comment by stefcameron** [comment]
@Slapbox 😆  It happens, I know! I'm still thankful that you ended-up on this issue because your inquiry led us to uncovering a bug that needs to be fixed, which will increase the quality and user satisfaction for this package.

It sounds like you've been able to sort things out, though, over on `focus-trap-react`, so I'm glad things are working for you now.

**Comment by stefcameron** [comment]
This is fixed in version 6.1.1 🎉 

@all-contributors add @Slapbox for bug
